src/solver/application.cpp

```diff
@@ -525,32 +525,30 @@ void Application::readDataForTheStudy(Data::StudyLoadOptions& options)
     initializeRandomNumberGenerators();
 }
 
-void Application::writeElapsedTime()
+void Application::writeInfoOnExecution()
 {
     if (!pStudy)
         return;
 
+    // Last missing duration to get : measure of total simulation duration
     pTotalTimer.stop();
-    pStudy->buffer.clear() << pStudy->folderOutput << Yuni::IO::Separator << "time_measurement.txt";
-    Benchmarking::CSVWriter writer(pStudy->buffer, &pBenchmarkingContentHandler);
-    // Write time data
-    writer.flush();
-}
+    pDurationCollector.addDuration("total", pTotalTimer.get_duration());
 
-void Application::writeStudyInfos()
-{
-    if (!pStudy)
-        return;
+    // Info collectors : they retrieve data from study and simulation
+    Benchmarking::StudyInfoCollector study_info_collector(*pStudy);
+    Benchmarking::SimulationInfoCollector simulation_info_collector(pOptimizationInfo);
 
-    // Collecting info from the study for a further record
-    Benchmarking::StudyInfoContainer study_info(*pStudy, pOptimizationInfo);
-    study_info.collect();
+    // Fill file content with data retrieved by collectors
+    Benchmarking::FileContent file_content;
+    pDurationCollector.toFileContent(file_content);
+    study_info_collector.toFileContent(file_content);
+    simulation_info_collector.toFileContent(file_content);
 
     // Flush previous info into a record file
-    Yuni::String filePath; 
-    filePath.clear() << pStudy->folderOutput << Yuni::IO::Separator << "study-info.txt";
-    Benchmarking::StudyInfoCSVwriter study_info_writer(filePath, study_info);
-    study_info_writer.flush();
+    Yuni::String filePath;
+    filePath.clear() << pStudy->folderOutput << Yuni::IO::Separator << "execution_info.ini";
+    Benchmarking::iniFilewriter ini_file_writer(filePath, file_content);
+    ini_file_writer.flush();
```

Comment:
The interface class `FileWriter` isn't very useful for now since `iniFileWriter` is its only child, and there is no need for dynamic polymorphism. Remove it ?

Reply:
Ok ! Let's keep it but then, make sure to use dynamic polymorphism in the application. Something like
```cpp
// application.h
std::unique_ptr<FileWriter> writer;
// application.cpp
writer = std::make_unique<iniFileWriter>(/* args */);
```

Also, I suggest renaming `FileWriter` to `IFileWriter`